src/CommunityToolkit.Maui.Core/Interfaces/IPopup.shared.cs

```diff
@@ -43,6 +43,11 @@ public interface IPopup : IElement, IVisualTreeElement
 	/// </summary>
 	LayoutAlignment VerticalOptions { get; }
 
+	/// <summary>
+	/// <see cref="TaskCompletionSource"/> that completes when the operating system has dismissed <see cref="IPopup"/> from the screen
+	/// </summary>
+	TaskCompletionSource PopupDismissedTaskCompletionSource { get; }
```

Comment:
Do we need it in interface?

Reply:
We do, because of hoe Handlers work. The handler only knows about `IPopup`; it doesn't no about `CommunityToolkit.Maui.Popup`. 

In order to access the `TaskCompletionSource` from the Handler, the only option is to include it in the `IPopup` interface. 

I agree with you though - putting a TCS in an Interface wouldn't be my first choice.

Reply:
How about creating a new interface just with the TCS, and a generic name, then IPopup extends this interface?

Reply:
Genius! Love it! 

I'll implement that now

Reply:
@pictos What do you think? https://github.com/CommunityToolkit/Maui/pull/1236